core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -3145,54 +3035,166 @@ protected function save($varValue)
 			}
 		}
 
+		// Make sure unique fields are unique
+		if ((\is_array($varValue) || (string) $varValue !== '') && ($arrData['eval']['unique'] ?? null) && !$this->Database->isUniqueValue($this->strTable, $this->strField, $varValue, $this->objActiveRecord->id))
+		{
+			throw new \Exception(sprintf($GLOBALS['TL_LANG']['ERR']['unique'], $arrData['label'][0] ?: $this->strField));
+		}
+
 		// Save the value if there was no error
-		if ((\is_array($varValue) || (string) $varValue !== '' || !($arrData['eval']['doNotSaveEmpty'] ?? null)) && ($this->varValue !== $varValue || ($arrData['eval']['alwaysSave'] ?? null)))
+		if ((\is_array($varValue) || (string) $varValue !== '' || !($arrData['eval']['doNotSaveEmpty'] ?? null)) && ($this->varValue != $varValue || ($arrData['eval']['alwaysSave'] ?? null)))
```

Comment:
This seems to be a behavior change. What are the implications?

Reply:
yes that is intentional. It should probably be fixed in 4.9 as well. In my test case, one of the values was `'0'` and the other was `0` (int vs. string), and the comparison failed to not save the value.

Reply:
@fritzmg No, it doesn‘t?

https://github.com/contao/contao/blob/c9fc13d208a481ea3997cb10314a393567223292/core-bundle/src/Resources/contao/drivers/DC_Table.php#L3138-L3140

Reply:
That is true. I noticed it because it always updated an empty select column.

Reply:
I think this need more investigation. 
For example array changes like `['some string'] != [true]` would then not be saved anymore. Although I’m not sure if such changes are actually possible here.
Also, didn’t we have issues in the past with storing `'0'` in fields that also accept the empty string? This could bring back this issue I think (`0 != ''`)

I’d rather convert the type to match the type in the database (as we do in #4800) and use a strict comparison then.
But please let’s move this into its own PR.

Reply:
> In my test case, one of the values was '0' and the other was 0 (int vs. string), and the comparison failed to not save the value.

As discussed, this is likely caused by PDO returning integers instead of strings in PHP 8.1. We should definitely debug this (also with regards to the `'' == 0` changes in PHP 8) and adjust the comparison accordingly.